lib/cocoapods/installer/pod_source_installer.rb

```diff
@@ -194,13 +190,10 @@ def released?
         !local? && !predownloaded? && sandbox.specification(root_spec.name) != root_spec
       end
 
-      def each_source_file(file_accessors, &blk)
-        file_accessors.each do |file_accessor|
-          file_accessor.source_files.each do |source_file|
-            next unless source_file.exist?
-            blk[source_file]
-          end
-        end
+      # @return [Array<Pathname>] The paths of the source files
+      #
+      def source_files(file_accessors)
+        file_accessors.flat_map(&:source_files)
```

Comment:
we could also cache this! I think `source_files` can be expensive within `file_accessor.rb`

Reply:
```ruby
@source_files ||= file_accessors.flat_map(&:source_files)
```

Reply:
I was going to cache this actually but there is code that inserts a `Dummy.swift` for app hosts that I _think_ broke when I did that. Need to look into it

Reply:
dummy swift was relatively new and I believe the validator does it only

Reply:
Nevermind, the broken test is the one I called out earlier where I stubbed `:lock_files!` and `:unlock_files!`

I think this is safe to cache